stl/inc/xstring

```diff
@@ -2946,9 +2947,10 @@ public:
             _Result._Res = _My_data._Myres + 1;
         } else {
             // use _BUF_SIZE + 1 to avoid SSO, if the buffer is assigned back
-            _Result._Ptr = _Al.allocate(_BUF_SIZE + 1);
+            size_type _Allocated = _BUF_SIZE + 1;
+            _Result._Ptr         = _Allocate_at_least_helper(_Al, _Allocated);
             _Traits::copy(_Unfancy(_Result._Ptr), _My_data._Bx._Buf, _BUF_SIZE);
-            _Result._Res = _BUF_SIZE + 1;
+            _Result._Res = _Allocated;

```

Comment:
Pre-existing question: What is going on here in this relatively new code? We're using the same value for the `allocate` and the `_Res`, but elsewhere `_Res` is always `1` smaller.

Reply:
I was convinced on Discord that I'm probably jumping at phantoms here.